src/mmw/apps/export/views.py

```diff
@@ -159,3 +163,50 @@ def shapefile(request):
     resp['Content-Disposition'] = 'attachment; '\
                                   'filename="{}.zip"'.format(filename)
     return resp
+
+
+@decorators.api_view(['POST'])
+def worksheet(request):
+    """Generate a ZIP of BMP Excel Worksheets prefilled with relevant data."""
+    # Extract list of pairs of HUC-12 + AoI
+    pairs = request.data
+
+    # Make a temporary directory to save the files in
+    tempdir = tempfile.mkdtemp()
+
+    for index, pair in enumerate(pairs):
+        worksheet_path = '{}/{}.xlsx'.format(tempdir, index)
+
+        # Copy the Excel template
+        shutil.copyfile(EXCEL_TEMPLATE, worksheet_path)
+
+        # Write Excel Worksheet
+        writer = BMPxlsx.Writer(worksheet_path)
+        # TODO Get this from the request body
+        writer.write({
+            'MMW Output': {
+                'L18': 96.66,
+                'L19': 10.09,
+                'L20': 86.57,
+            }
+        })
+        writer.close()
+
+    worksheets = glob.glob('{}/*.xlsx'.format(tempdir))
+
+    # Create a zip file in memory for all the worksheets
+    stream = StringIO.StringIO()
+    with zipfile.ZipFile(stream, 'w') as zf:
+        for fpath in worksheets:
+            _, fname = os.path.split(fpath)
+            zf.write(fpath, fname)
+            os.remove(fpath)
```

Comment:
Do you have a sense of the memory consumption of these files in a worst case scenario? Curious to see what kind of load we could expect that would put memory pressure on the app servers. I expect the worksheet is relatively small, and that the likely number of intersected HUCs wouldn't exceed single digits.

If we have access to a logger, it would be interesting to `DEBUG` the number files created in the zip so we can monitor use metrics.

Reply:
I'll make a card to add some monitoring. Also to see if we can limit the size of the smaller area of interest, which should protect from being too large and intersecting too many HUC-12s.